polkadot/node/network/bridge/src/rx/mod.rs

```diff
@@ -551,6 +551,14 @@ where
 	let mut peers = Vec::with_capacity(neighbors.len());
 	for (discovery_id, validator_index) in neighbors {
 		let addr = get_peer_id_by_authority_id(ads, discovery_id.clone()).await;
+		if addr.is_none() {
+			// See on why is not good in https://github.com/paritytech/polkadot-sdk/issues/2138
+			gum::warn!(
```

Comment:
```suggestion
			gum::debug!(
```

Or what should a node operator do when they see this message?

Reply:
Unfortunately, not much they can do, we need to actually fix this potential bug one way or another.

I was thinking of having it as WARN, because this increases the chance of this message making it back to us and gives us a datapoint if we should prioritise the fix or if it is just theoretical.

To that end I can probably put the ticket in the log message. 

Putting it as `debug` wouldn't help us in any way here.

Reply:
Wouldn't we get a ton of warnings on startup because of that?
https://github.com/paritytech/polkadot-sdk/blob/dca142398e00786429d69eb8b9e4454499962b2a/substrate/client/authority-discovery/src/lib.rs#L102

Reply:
> I was thinking of having it as WARN, because this increases the chance of this message making it back to us and gives us a datapoint if we should prioritise the fix or if it is just theoretical.

This is a very bad idea on getting this kind of data. The only thing that will happen is that validator operators will be confused by the warning and not knowing what to do. 

I also don't get the entire logic here. If you know that there is a bug, then fix it. If you need to find out how regularly it is happening, can you not see it in the debug logs of versi?

Reply:
> This is a very bad idea on getting this kind of data. The only thing that will happen is that validator operators will be confused by the warning and not knowing what to do.

Isn't that better than something bad happened and we have no idea that it happened. 

> I also don't get the entire logic here. If you know that there is a bug, then fix it. If you need to find out how regularly it is happening, can you not see it in the debug logs of versi?

I was following the logic where if  this is happening we would like to know and if it doesn't happen anywhere maybe it doesn't need fixing or prioritising. Adding the log line was cheap it wasn't meant to replace the fixing.

I don't think anyone reads the debug logs unless they actively debug something, so they would be lost.

Reply:
> Wouldn't we get a ton of warnings on startup because of that?
> 
> https://github.com/paritytech/polkadot-sdk/blob/dca142398e00786429d69eb8b9e4454499962b2a/substrate/client/authority-discovery/src/lib.rs#L102

That's a good point, would have to look closer.

Reply:
> Isn't that better than something bad happened and we have no idea that it happened.

Yeah for sure we should log if something bad is happening. However, the possibility that you can not resolve a validator at some point in time is very high. If the user (node operator) can not fix this, there is no need to spam them with this information.

Reply:
>If the user (node operator) can not fix this, there is no need to spam them with this information

I see your point, changed it to debug.